src/entities/user.entity.ts

```diff
@@ -213,6 +215,26 @@ async function getUserByDID(did: string): Promise<Result<UserEntity, GetUserErr>
 	}
 }
 
+async function deleteUserByDID(did: string): Promise<Result<void, DeleteUserErr>> {
+	try {
+		const user = await userRepository.createQueryBuilder('user')
+			.where("user.did = :did", { did: did })
+			.getOne();
+		await webauthnCredentialRepository.delete({
+			user: { id: user.id }
+		});
+
+		await userRepository.delete({
+			did: did
+		});
```

Comment:
Perhaps these should be done atomically in a transaction, like we do in [`deleteWebauthnCredential`](https://github.com/wwWallet/wallet-backend-server/pull/46/files#diff-dbee3db7cbeaa6761791c04564c60bfa1309c83f5e2a9754a5206aa091a71efeR412) for example?

Reply:
I agree. I just updated the specific function to send one transaction for the three sql commands. Let me know if there is anything that needs to be changed.

Reply:
Nice :+1: I guess it could still happen that `deleteAllFcmTokensForUser` succeeds and `deleteUserByDID` fails, but maybe that's not as big a problem?